src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -395,26 +395,39 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
             return openDocs;
         }
 
-        internal Task OnDocumentOpenedAsync(
+        protected internal void OnDocumentOpened(
             DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
-            OnDocumentOpened(documentId, textContainer, isCurrentContext);
-            return SpecializedTasks.EmptyTask;
+            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: true).Wait(CancellationToken.None);
         }
 
-        protected internal void OnDocumentOpened(
+        internal Task OnDocumentOpenedAsync(
```

Comment:
`OnDocumentOpenedAsync` only seems to be reference by tests at the moment, which is a little scary that all of our tests go through the async path, but all of the productions scenarios go through the sync path.  Is the plan to change the product code in the near future?  If not, I would prefer to remove the async path.